src/engine/strat_engine/pool.rs

```diff
@@ -281,28 +281,35 @@ impl Pool for StratPool {
         pool_uuid: PoolUuid,
         pool_name: &str,
         specs: &[(&'b str, Option<Sectors>)],
-    ) -> StratisResult<Vec<(&'b str, FilesystemUuid)>> {
+    ) -> StratisResult<SetCreateAction<(&'b str, FilesystemUuid)>> {
         let names: HashMap<_, _> = HashMap::from_iter(specs.iter().map(|&tup| (tup.0, tup.1)));
-        for name in names.keys() {
-            validate_name(name)?;
-            if self.thin_pool.get_mut_filesystem_by_name(*name).is_some() {
-                return Err(StratisError::Engine(
-                    ErrorEnum::AlreadyExists,
-                    name.to_string(),
-                ));
-            }
-        }
 
         // TODO: Roll back on filesystem initialization failure.
         let mut result = Vec::new();
+        let mut unchanged_result = Vec::new();
         for (name, size) in names {
-            let fs_uuid = self
-                .thin_pool
-                .create_filesystem(pool_uuid, pool_name, name, size)?;
-            result.push((name, fs_uuid));
+            if self.thin_pool.get_mut_filesystem_by_name(name).is_none() {
```

Comment:
For this one, I chose expect because it should be unreachable. I've seen a number of places where expect is used in the codebase for unreachable code paths and wanted to be consistent with the rest of the coding style. This also seems shorter than the alternative which, for me, would be:
```rust
let uuid = match self.thin_pool.get_filesystem_by_name(name) {
    Some((uuid, _)) => uuid,
    None => unreachable!("Filesystem must be in thinpool to reach this code"),
}
```
Does that seem reasonable?

Reply:
Changed to `unreachable!()`.